### ISSUE TITLE
NewType is not assignable to TypeForm

### ISSUE THREAD

**Issue by danielknell** [microsoft/pyright#11328]
**Describe the bug**

A NewType type can be passed to `TypeForm` parameter, but passing a value explicitly typed as NewType raises an error.

```
Argument of type "type | NewType" cannot be assigned to parameter "t" of type "TypeForm[Unknown]" in function "unwrap"
  Type "type | NewType" is not assignable to type "TypeForm[Unknown]"
    "NewType" is not assignable to "TypeForm[Unknown]"  (reportArgumentType)
```

If NewType's are not valid for TypeForm, I would expect the below to fail on line 7 and 12, and if they are supported, I would expect neither to error. Instead it only errors on line 7.

mypy does not report any errors.

**Code or Screenshots**

```python
import typing_extensions as typing

X = typing.NewType("X", int)

def unwrap(t: typing.TypeForm) -> typing.TypeForm:
    if isinstance(t, typing.NewType):
        return unwrap(t.__supertype__)

    return t

typing.reveal_type(X)
typing.reveal_type(unwrap(X))
```

[pyright playground](https://pyright-play.net/?enableExperimentalFeatures=true&code=JYWwDg9gTgLgBDAnmYA7A5gfQKYA8baoDOwExcAhkQsmugFD0AacAvDShgHQBy2A7gBVk2ABQAiJuIA0cNDACUjACbYAZnACuqflAphRMAFwc6XYWGwAxaCAVwAtAD5T3C9dtH6cH3I3ASYhgKVABjMRhZJE50XgF3BS9fZLgobBhNKFQtHT0DGC5MTCJNS1gRIqVvXzSMrIRGaLM0gDdsCgAbTGixJiUm7lb2rp7RbV19UT6FIA)

**VS Code extension or command-line**

pyright playground using version 1.1.408

**Comment by erictraut** [comment]
I think pyright is correct here. The expression `t.__supertype__` is not of type `TypeForm`, nor is it a valid type expression. Therefore it is not assignable to `TypeForm`.

For details about how `TypeForm` works and its assignability rules, refer to [PEP 747](https://peps.python.org/pep-0747/).

**Comment by danielknell** [comment]
@erictraut there are other areas where this causes issue aside from __supertype__.

```py
import typing_extensions as typing

class Visitor(typing.Protocol):
    def __call__(self, t: typing.TypeForm, new_type: bool) -> None: ...


def walk_types(types: tuple[typing.TypeForm, ...], visit: Visitor) -> None:
    for t in types:
        if isinstance(t, typing.NewType):
            visit(t, True)
        else:
            visit(t, False)
```

also fails the same as it narrows the type of `t` to `NewType` on line 10 instead of `NewType & TypeForm[Unknown]`, therefore a value that would otherwise be allowed now raises an error.

does narrowing to the concrete type instead of an intersection imply `NewType` is considered assigneable to `TypeForm`?


```py
import typing_extensions as typing

class Visitor(typing.Protocol):
    def __call__(self, t: typing.TypeForm) -> None: ...


def walk_types(types: tuple[typing.TypeForm, ...], visit: Visitor) -> None:
    for t in types:
        if isinstance(t, typing.NewType):
            print("new type")

        visit(t)
```

also fails for the same reason because `t` is `TypeForm[Unknown] | NewType` on line 12.
